### PR TITLE
Update `SplFixedArray` methods + add PHP 8.2.0 changelog entry

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix https://github.com/php/doc-en/issues/1702

### CODE REVIEW

**Review by Girgias — CHANGES_REQUESTED** [review]
The documentation documents both PHP 7 and 8 so removing the deprecated methods is a no go.

Has the version.xml file been updated to indicate they have been deprecated and removed? As that would help a bit.

**Comment by alexandre-daubois** [comment]
Thanks for the tip about `versions.xml`. Does it look better now?

File: reference/spl/versions.xml [hunk 1]
```diff
@@ -933,11 +933,15 @@
 
  <function name="splfixedarray" from="PHP 5 &gt;= 5.3.0, PHP 7, PHP 8"/>
  <function name="splfixedarray::__construct" from="PHP 5 &gt;= 5.3.0, PHP 7, PHP 8"/>
+ <function name="splfixedarray::__serialize" from="PHP 8 &gt;= 8.2.0"/>
+ <function name="splfixedarray::__unserialize" from="PHP 8 &gt;= 8.2.0"/>
```
Comment (Girgias) [inline reference/spl/versions.xml:937]: Are these just implemented as of 8.2?

File: reference/spl/versions.xml [hunk 2]
```diff
@@ -933,11 +933,15 @@
...
  <function name="splfixedarray::__wakeup" from="PHP 5 &gt;= 5.5.0, PHP 7, PHP 8"/>
  <function name="splfixedarray::count" from="PHP 5 &gt;= 5.3.0, PHP 7, PHP 8"/>
  <function name="splfixedarray::toarray" from="PHP 5 &gt;= 5.3.0, PHP 7, PHP 8"/>
  <function name="splfixedarray::fromarray" from="PHP 5 &gt;= 5.3.0, PHP 7, PHP 8"/>
+ <function name="splfixedarray::getIterator" from="PHP 8"/>
  <function name="splfixedarray::getsize" from="PHP 5 &gt;= 5.3.0, PHP 7, PHP 8"/>
+ <function name="splfixedarray::jsonSerialize" from="PHP 8 &gt;= 8.1.0"/>
```
Comment (Girgias) [inline reference/spl/versions.xml:944]: Ditto

**Review by Girgias — COMMENTED** [review]
@kocsismate can you have a quick look?

File: reference/spl/splfixedarray/unserialize.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8"?>
+<refentry xml:id="splfixedarray.unserialize" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <refnamediv>
+  <refname>SplFixedArray::__unserialize</refname>
+  <refpurpose>Deserializes the <parameter>data</parameter> parameter into a SplFixedArray object</refpurpose>
```
Comment (kocsismate) [inline reference/spl/splfixedarray/unserialize.xml:5]:
```suggestion
  <refpurpose>Deserializes the <parameter>data</parameter> parameter into an SplFixedArray object</refpurpose>
```

File: reference/spl/versions.xml (hunk 1 above)
Comment (alexandre-daubois) [reply reference/spl/versions.xml:937]: I checked the reflection, and yes they came up in 8.2 🙂

File: reference/spl/versions.xml (hunk 2 above)
Comment (alexandre-daubois) [reply reference/spl/versions.xml:944]: Same, it came up in 8.1 👍

File: reference/spl/splfixedarray/unserialize.xml (hunk 3 above)
Comment (alexandre-daubois) [reply reference/spl/splfixedarray/unserialize.xml]: Updated both, thanks for the review!

File: reference/spl/splfixedarray.xml [hunk 4]
```diff
@@ -72,6 +72,13 @@
       </row>
      </thead>
      <tbody>
+      <row>
+       <entry>8.2.0</entry>
+       <entry>
+        <function>__serialize</function> and <function>__unserialize</function>
+        magic methods are added to <classname>SplFixedArray</classname>.
```
Comment (Girgias) [inline reference/spl/splfixedarray.xml:79]:
```suggestion
        The <methodname>SplFixedArray::__serialize</methodname> and <methodname>SplFixedArray::__unserialize</methodname>
        magic methods have been added to <classname>SplFixedArray</classname>.
```

Or something along those lines. :)

**Review by Girgias — APPROVED** [review]
One comment and then good to land

File: reference/spl/splfixedarray.xml (hunk 4 above)
Comment (alexandre-daubois) [reply reference/spl/splfixedarray.xml]: Updated, that's better :+1:

**Pull request merged**